src/jit/SimdArm32Inl.h

```diff
@@ -267,25 +590,161 @@ static void simdEmitI32x4Dot(sljit_compiler* compiler, sljit_s32 vd, sljit_s32 v
+        sljit_emit_fop1(compiler, op, dst.arg, dst.argw + (i * sizeof(double)), SLJIT_FR0, 0);
+    }
+}
+
+static void simdEmitF64x2Unary(sljit_compiler* compiler, JITArg src, JITArg dst, uint32_t op)
+{
+    for (uint8_t i = 0; i < 2; i++) {
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR0, src.arg, src.argw + (i * sizeof(double)));
+        sljit_emit_op_custom(compiler, &op, sizeof(op));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_STORE, SLJIT_FR0, dst.arg, dst.argw + (i * sizeof(double)));
+    }
+}
+
+static void simdEmitF64x2Binary(sljit_compiler* compiler, JITArg src[2], JITArg dst, sljit_s32 op)
+{
+    for (uint8_t i = 0; i < 2; i++) {
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR0, src[0].arg, src[0].argw + (i * sizeof(double)));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR1, src[1].arg, src[1].argw + (i * sizeof(double)));
+        sljit_emit_fop2(compiler, op, dst.arg, dst.argw + (i * sizeof(double)), SLJIT_FR0, 0, SLJIT_FR1, 0);
+    }
+}
+
+static void simdEmitF64x2CMP(sljit_compiler* compiler, JITArg src[2], JITArg dst, sljit_s32 opcode, sljit_s32 flag)
+{
+    opcode |= SLJIT_CMP_F64;
+    auto tmpReg = SLJIT_R0;
+
+    for (uint8_t i = 0; i < 2; i++) {
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_LOAD | SLJIT_MEM_UNALIGNED, SLJIT_FR0, src[0].arg, src[0].argw + (i * sizeof(double)));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_LOAD | SLJIT_MEM_UNALIGNED, SLJIT_FR1, src[1].arg, src[1].argw + (i * sizeof(double)));
+        sljit_emit_fop1(compiler, opcode, SLJIT_FR0, 0, SLJIT_FR1, 0);
+        sljit_emit_op_flags(compiler, SLJIT_MOV, tmpReg, 0, flag);
+        sljit_emit_op2(compiler, SLJIT_SUB, tmpReg, 0, SLJIT_IMM, 0, tmpReg, 0);
+        sljit_emit_mem(compiler, SLJIT_MOV | SLJIT_MEM_STORE | SLJIT_MEM_UNALIGNED, tmpReg, dst.arg, dst.argw + ((i * 2) * sizeof(float)));
+        sljit_emit_mem(compiler, SLJIT_MOV | SLJIT_MEM_STORE | SLJIT_MEM_UNALIGNED, tmpReg, dst.arg, dst.argw + (((i * 2) + 1) * sizeof(float)));
+    }
+}
+
+static void simdEmitF64x2Ne(sljit_compiler* compiler, JITArg src[2], JITArg dst)
+{
+    auto tmpReg = SLJIT_R0;
+
+    for (uint8_t i = 0; i < 2; i++) {
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR0, src[0].arg, src[0].argw + (i * sizeof(double)));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR1, src[1].arg, src[1].argw + (i * sizeof(double)));
+        sljit_emit_fop1(compiler, SLJIT_CMP_F64 | SLJIT_SET_ORDERED_EQUAL, SLJIT_FR0, 0, SLJIT_FR1, 0);
+        sljit_emit_op_flags(compiler, SLJIT_MOV, tmpReg, 0, SLJIT_ORDERED_EQUAL);
+        sljit_emit_op2(compiler, SLJIT_ADD, tmpReg, 0, SLJIT_IMM, -1, tmpReg, 0);
```

Comment:
This is a complex operation. `SUB 0 - reg` is usually supported. The flag can be inverted for  sljit_emit_op_flags.